ios/brave-ios/Sources/AIChat/ModelView/AIChatModelView.swift

```diff
@@ -0,0 +1,303 @@
+    
+    Task { @MainActor in
+      await getPremiumStatus()
+    }
+    
+    if isAgreementAccepted {
+      api.setConversationActive(true)
+    }
+  }
+  
+  public func getPageTitle() -> String? {
+    if let webView = webView {
+      // Return the Page Title
+      if let title = webView.title, !title.isEmpty {
+        return title
+      }
+      
+      guard let url = getLastCommittedURL() else {
+        return nil
+      }
+      
+      // Return the URL domain/host
+      if url.pathExtension.isEmpty {
+        return URLFormatter.formatURLOrigin(forDisplayOmitSchemePathAndTrivialSubdomains: url.absoluteString)
+      }
+      
+      // Return the file name with extension
+      return url.lastPathComponent
+    }
+    return nil
+  }
+  
+  public func getLastCommittedURL() -> URL? {
+    if let url = webView?.url {
+      return InternalURL.isValid(url: url) ? nil : url
+    }
+    return nil
+  }
+  
+  public func getPageContent(completion: @escaping (String?, Bool) -> Void) {
+    guard let webView = webView else {
+      completion(nil, false)
+      return
+    }
+    
+    requestInProgress = true
+    
+    Task { @MainActor in
+      defer { requestInProgress = false }
+      
+      if await script.getPageContentType(webView: webView) == "application/pdf" {
+        if let base64EncodedPDF = await script.getPDFDocument(webView: webView) {
+          completion(await AIChatPDFRecognition.parse(pdfData: base64EncodedPDF), false)
+          return
+        }
+        
+        // Could not fetch PDF for parsing
+        completion(nil, false)
+        return
+      }
+      
+      // Fetch regular page content
+      let text = await script.getMainArticle(webView: webView)
+      if let text = text, !text.isEmpty {
+        completion(text, false)
+        return
+      }
+      
+      // No article text. Attempt to parse the page as a PDF/Image
+      let render = UIPrintPageRenderer()
+      render.addPrintFormatter(webView.viewPrintFormatter(), startingAtPageAt: 0)
+      
+      let page = CGRect(x: 0, y: 0, width: 595.2, height: 841.8) // A4, 72 dpi
+      let printable = page.insetBy(dx: 0, dy: 0)
+      
+      render.setValue(NSValue(cgRect: page), forKey: "paperRect")
+      render.setValue(NSValue(cgRect: printable), forKey: "printableRect")
+      
+      // 4. Create PDF context and draw
+      let pdfData = NSMutableData()
+      UIGraphicsBeginPDFContextToData(pdfData, CGRect.zero, nil)
+      for i in 0..<render.numberOfPages {
+        UIGraphicsBeginPDFPage()
+        let bounds = UIGraphicsGetPDFContextBounds()
+        render.drawPage(at: i, in: bounds)
+      }
+      
+      UIGraphicsEndPDFContext()
+      completion(await AIChatPDFRecognition.parseToImage(pdfData: pdfData.base64EncodedString()), false)
+    }
+  }
+  
+  public func isDocumentOnLoadCompletedInPrimaryFrame() -> Bool {
+    return webView?.isLoading == false
+  }
+  
+  public func onHistoryUpdate() {
```

Comment:
Please MARK or separate out the AIChatDelegate methods so its clear what is what